From 7504f3440c7406060838b3a969f1926a886505ca Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 12 Dec 2018 19:36:27 +0200
Subject: [PATCH 0827/1051] net: mvpp2: fix cls flow uninitialized loop
 variable

Local loop's variable is used uninitialized in construction
 while(!is_last){ is_last = setting(); }.
As result the initialization loop is aborted.

Fix:
Use construction do{ is_last = setting }while(!is_last);

NOTE: the original "uninitialized" construction is not detected
by GCC-Linaro-7.3

Change-Id: I785e52539b1a7760323d68a25eb4bdc254af7de9
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61727
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
index 30af059768ef..16018ed94b3e 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
@@ -628,7 +628,7 @@ int mvpp2_cls_flow_hash_find(struct mvpp2_port *port,
 	int engine, is_last, flow_offset, port_bm, idx = 0;
 
 	flow_offset = 0;
-	while (!is_last) {
+	do {
 		idx = MVPP2_PORT_FLOW_INDEX(flow_offset, flow->flow_id);
 		mvpp2_cls_flow_read(port->priv, idx, fe);
 		engine = mvpp2_cls_flow_eng_get(fe);
@@ -639,7 +639,7 @@ int mvpp2_cls_flow_hash_find(struct mvpp2_port *port,
 		    (port_bm & BIT(port->id)))
 			break;
 		flow_offset++;
-	}
+	} while (!is_last);
 
 	*flow_index = idx;
 	if (is_last)
-- 
2.17.1

